### ISSUE TITLE
PSA: Batch file to find all Java files

### ISSUE THREAD

**Issue by aquohn** [nusCS2113-AY1920S1/forum#34]
I spent about an hour today wrangling with Windows' cmd/batch file syntax so you don't have to.

This is specifically for the case where your classes have been put into packages, and hence into folders, and therefore compilation from the command line is no longer straightforward ("cannot find symbol" errors galore). This code snippet can be dropped into your `runtest.bat` batch file, and the old compile command removed, although you should check the paths.

```
REM create a new file listing all your Java sources
type nul > all.javas
REM find all Java source files
for /F "tokens=*" %%f in ('dir ..\src\main\java /B/A:-D/S "*.java" 2^>nul') do (
    echo %%f >> all.javas
)

javac  -cp ..\bin\production -Xlint:none -d ..\bin\production @all.javas
```

Basically, this finds all the Java files in your source directory, puts them in a file, and feeds the file to `javac`, which then compiles each file into a class. `-cp` specifies where to look for classes and sources (if you're using it just to find sources, use `-sourcepath` instead for greater clarity), while `-d` specifies where the compiled class files should go. `-cp` is actually redundant for my build, but if you have any other classes that your files are dependent on, they should be in the `-cp` folder.

For those intending to do a Makefile/cmake build - don't, `javac` is weird in that it takes way longer to start the individual processes than it takes to recompile everything at one shot.

**Comment by Jefferson111** [comment]
MVP
